src/_ert_forward_model_runner/job.py

```diff
@@ -124,11 +131,40 @@ def ensure_file_handles_closed():
                 # state, we ignore any failures. Only seen on OSX thus far.
                 #
                 # See https://github.com/giampaolo/psutil/issues/1044#issuecomment-298745532  # noqa
-                memory = 0
-            if memory > max_memory_usage:
-                max_memory_usage = memory
+                memory_rss = 0
 
-            yield Running(self, max_memory_usage, memory)
+            max_memory_usage = max(memory_rss, max_memory_usage)
+
+            oom_score = None
+            # oom_score defaults to 0 in Linux, but varies between -1000 and 1000.
+            # A value of None means that we have no information.
+            with contextlib.suppress(ValueError, FileNotFoundError):
+                oom_score = int(
+                    Path(f"/proc/{process.pid}/oom_score").read_text(encoding="utf-8")
+                )
+            with contextlib.suppress(NoSuchProcess, AccessDenied, ZombieProcess):
+                for child in process.children(recursive=True):
+                    with contextlib.suppress(ValueError, FileNotFoundError):
+                        oom_score_child = int(
+                            Path(f"/proc/{child.pid}/oom_score").read_text(
+                                encoding="utf-8"
+                            )
+                        )
+                        if oom_score is None:
+                            oom_score = oom_score_child
+                        else:
+                            oom_score = max(oom_score, oom_score_child)
```

Comment:
I think this would benefit from a comment. I thought the score for the main process is computed from the children processes automatically.

Reply:
Hmm, do you have a reference for this? I don't think it can be the case, because then the mother-of-all processes with pid 1 would always have the highest score and get killed.

Reply:
Based on the chatgpt, you are right. Also agree with the `max` function.